pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -114,6 +116,19 @@ pub struct IndexPart {
     /// The timestamp when the timeline was marked invisible in synthetic size calculations.
     #[serde(skip_serializing_if = "Option::is_none", default)]
     pub(crate) marked_invisible_at: Option<NaiveDateTime>,
+
+    /// The encryption key used to encrypt the timeline layer files.
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub(crate) keys: Option<Vec<EncryptionKey>>,
+}
+
+#[serde_as]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct EncryptionKey {
+    #[serde_as(as = "Base64")]
+    pub key: Vec<u8>,
+    pub version: u32,
```

Comment:
I think we should have a newtype for the version similar to how we have one for the generation etc.

Also, the version should at least be a tuple of `(generation, version)`.